Documentation/design/neutron_plugin_for_mizar.md

```diff
@@ -0,0 +1,27 @@
+# Neutron plugin for mizar
+ 
+Openstack is one of the most popular opensource cloud platform, of which the network function subsystem is neutron. This Document is about to explan how to design a neutron plugin for mizar, in order that compute resources provided by openstack could use the mizar network pipeline. 
+
+neutron plugin might include two major functions: resource model save and transformation and network config distribution.
+
+## Resource model save and transformation
+
+The plugin should provide all operation APIs for mizar resources(vpc, network, endpoint, etc.), as the resource models of mizar are not same as neutron(router, network, subnet, port, etc.). Besides, neutron is one of the core subsystems of openstack, other core subsystems like nova may couple with neutron models. In order to make other subsystems of openstack like nova could properly interaction with neutron, when to create a mizar resource we also need to create one or more neutron resources corrspondingly. Those co-created neutron resources should only be stored, as the original neutron agents may not be needed any more and the mizar agents(transit agent, etc.) might only concern about the mizar resources.
+
+![neutron plugin resource model map](./figures/neutron_plugin_model_map.png)

```

Comment:
The mapping should be:

Mizar -> Neuton
VPC -> Network
Network -> Subnet
Simple Endpoint -> Port

Reply:
I get a little bit confused here. If a VPC mapping to a network means that all networks (Mizar) inside a VPC shared one boradcast zone? Only those crossing traffics between VPC need be sent to a  transit router (divider)?

Reply:
Good question. Best explained by example.

Let's say we have a VPC with CIDR block 10.0.0.0/12 (This is a network in neutron)
We can have two networks below the VPCs net1: 10.1.0.0/24 and net2: 10.20.0.0/24 (These are subnets in neutron)

1. Traffic from net1 to net2 has to go through the divider
1. Traffic within net1 or within net2 does not go through the divider (only through the bouncer once and fast-path is then direct between endpoints)
1. If we have another VPC, then traffic between the two VPCs will go through the divider

As of broadcast/multicast traffic: we do not support them right now. We can discuss the real use cases in our next sprint planning. I would say broadcast/multicast may be supported within a network boundary (bouncer).

Note: Mizar data-plane does not work as traditional networks, for example, if we have an ARP request the bouncer does not broadcast it to all endpoints, because it knows where the target endpoint is. With this, a network is not necessarily a subnet within the VPC. It can be any arbitrary group of endpoints. 

Does that make sense? Again, we would love to discuss use cases of broadcast/multicast in the upcoming sprint planning.